### PR TITLE
Cortex basic auth

### PR DESCRIPTION
**Pull request by mleahu**
### Description of your changes
Implement the basic auth to cortex API.

I have:

- [ ] Read and followed Crossplane's [contribution process].
- [X] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

See examples for provider configuration and test managed resources.

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: examples/provider/config.yaml [hunk 1]
```diff
@@ -1,7 +1,23 @@
...
+      key: credentials
+---
+# example secret with credentials {"username": "your_username", "password": "your_password"} in base64 encode
+apiVersion: v1
+kind: Secret
+metadata:
+  name: secret-name
+  namespace: crossplane-system
+data:
+  credentials: eyJ1c2VybmFtZSI6ICJ5b3VyX3VzZXJuYW1lIiwgInBhc3N3b3JkIjogInlvdXJfcGFzc3dvcmQifQ==
```
Comment (janwillies) [inline examples/provider/config.yaml:23]: What do you think about having two keys, `user` and `password`, instead of a json encoded one in `credentials`?

Comment (mleahu) [reply examples/provider/config.yaml:23]: this depends on how we will get the secret from AWS Secret Manager, so might be subject to change. As of now, the Service Catalog products create the secrets like that.

Comment (mleahu) [reply examples/provider/config.yaml:28]: I updated the PR with parameterization of secret keys that correspond to the cortex configurations for api user, key or auth token.

**Comment by teeverr** [comment]
Looks good : )

**Review by teeverr — APPROVED** [review]

**Review by stefnans — APPROVED** [review]

**Pull request merged**